### PR TITLE
Feat: URL for expanded row

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Review by dstala — APPROVED** [review]
@pranavxc 
- ~~Consider enabling the `comment` section by default in expanded record drawer~~ This will slow it down
- ~~Consider showing `hidden columns` as discussed (separate PR may be)~~ Ya can be later
- If `row ID` is invalid, skip opening `expand record` draw

File: packages/nc-gui/components/smartsheet/Gallery.vue [hunk 1]
```diff
@@ -18,6 +18,7 @@ import {
 } from '#imports'
 import Row from '~/components/smartsheet/Row.vue'
 import type { Row as RowType } from '~/composables'
+import { extractPkFromRow } from '~/utils'
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Gallery.vue:21]: -> `'#imports'`

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import type { ColumnType } from 'nocodb-sdk'
 import { UITypes, isVirtualCol } from 'nocodb-sdk'
 import { message } from 'ant-design-vue'
+import { useRoute } from '#app'
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Grid.vue:5]: -> `'#imports'`

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 3]
```diff
@@ -34,6 +35,7 @@ import {
 } from '#imports'
 import type { Row } from '~/composables'
 import { NavigateDir } from '~/lib'
+import { extractPkFromRow } from '~/utils'
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Grid.vue:38]: -> '#imports'

File: packages/nc-gui/components/smartsheet/expanded-form/index.vue [hunk 4]
```diff
@@ -1,11 +1,13 @@
 <script setup lang="ts">
+import { message } from 'ant-design-vue'
 import type { TableType, ViewType } from 'nocodb-sdk'
 import { UITypes, isSystemColumn, isVirtualCol } from 'nocodb-sdk'
 import type { Ref } from 'vue'
 import Cell from '../Cell.vue'
 import VirtualCell from '../VirtualCell.vue'
 import Comments from './Comments.vue'
 import Header from './Header.vue'
+import { useRouter } from '#app'
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/expanded-form/index.vue:10]: -> '#imports'

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

**Review by wingkwong — COMMENTED** [review]
1. sanitise `route.query.rowId`? 
2. should the pagination also be changed when I use this URL? For example, rowId = 45 (the 20th record on page 2 with size page 25), then the page should be navigated to Page 2. Currently it's always page 1.

**Review by wingkwong — APPROVED** [review]

**Pull request merged**